### PR TITLE
ref(modals): Improve integrationCodeMappings modal

### PR DESCRIPTION
**Pull request by evanpurkhiser**
- Use openModal instead of maintaining 'modal is opened' state
 - Improve aesthetics of the modal
 - Improve aesthetics of the IntegrationCodeMappings panel header

Old:
image
image

New:
image
image

I switched out the JsonForm for individual `FieldFromConfig`s, and then set the `stacked` and `inline={false}` properties for the improved modal form style. I also added `flexibleControlStateSize` to avoid extra space on the right unless there is an error.

### CODE REVIEW

**Comment by EvanPurkhiser** [comment]
Fixed up the tests on this. Should be good to go now.

File: src/sentry/static/sentry/app/views/settings/organizationIntegrations/configureIntegration.tsx [hunk 1]
```diff
@@ -206,7 +206,7 @@ class ConfigureIntegration extends AsyncView<Props, State> {
   renderMainContent(provider: IntegrationProvider) {
     const {integration} = this.state;
     //if no code mappings, render the single tab
-    if (!this.hasStacktraceLinking(provider)) {
+    if (this.hasStacktraceLinking(provider)) {
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/settings/organizationIntegrations/configureIntegration.tsx:209]: whoops

Comment (scefali) [reply src/sentry/static/sentry/app/views/settings/organizationIntegrations/configureIntegration.tsx:209]: @EvanPurkhiser yea this broke everything hahaha

**Comment by priscilawebdev** [comment]
@evanh it looks great! Good Job 🚀 could we maybe just use the `closeButton` prop in the modal's header?

**Review by MeredithAnya — APPROVED** [review]

**Pull request merged**